python/src/tstools/inputFileIO.py

```diff
@@ -139,7 +141,7 @@ def readFitFile(self, fileName):
         
         # check that if a non-linear method is chosen that a
         # local minimum finder is also chosen
-        if self.im != 'linear' and self.lm == '':
+        if (self.im != 'linear' and self.im != 'gensyn') and self.lm == '':
```

Comment:
This `if` statement might be a good canditate for a helper method

```python
def _validStuffChosen
  return self.im != 'linear' and self.im != 'gensyn' and self.lm == ''
```

```python
if self._validStuffChosen():
```

Same for the big if statement below